### PR TITLE
Bulk merge of docs from 'docs' to 'master'

### PR DESCRIPTION
**Pull request by jackiekazil**
This is a bunch of doc updates that are coming into master. These have already been reviewed, so I will do a bulk merge after the last one that I am expecting comes in:  #668, #669, #671, #672, #674, #675, #677, #682, #683, ...

### CODE REVIEW

**Comment by jackiekazil** [comment]
#683 is still pending. Waiting for a code block fix in docs.

**Comment by Corvince** [comment]
Is this really supposed to revert #651 ?

**Comment by jackiekazil** [comment]
Last PR waiting to be merged is #686 into docs, then I can merge this all into master.

**Comment by Corvince** [comment]
It looks like this is reverting #651 ?? Is this correct?

**Comment by jackiekazil** [comment]
@Corvince good catch!! That happened because of a bug issue that happened in with the git merges. Thoughts on how to have navigate? 

I am thinking of cherry picking those commits into another branch, and then merging that branch into this one.... basically recreating that PR. 

As a result, I feel like you or someone else should do a review of that part at the very least to make sure that "looks good" before this goes into master. :sigh:

**Comment by Corvince** [comment]
Hey, not sure if there have been useful bug-fixing changes to these files (not clear from your text), but you could just checkout the files from the commit before the changes and then commit to this branch:

`git checkout fa7bbb0 -- mesa/batchrunner.py tests/test_batchrunner.py`

**Comment by jackiekazil** [comment]
@Corvince looking into this.

**Comment by jackiekazil** [comment]
After the merge by @Corvince of #691. I believe this PR is ready to merge. Since I missed the thing that @Corvince caught, I would like a second pair of eyes on this to give it a glance over. I already gave these commits a first review when merging into this branch. I am being extra careful.

File: docs/tutorials/intro_tutorial.rst [hunk 1]
```diff
@@ -48,27 +48,42 @@ Let's get started.
...
-Python3 and does not work in Python 2 environments.
+To start, install Mesa. We recommend using a `pipenv <https://docs.pipenv.org/en/latest/>`_,
+which combines the `virtual
+environment <https://virtualenvwrapper.readthedocs.org/en/stable/>`_
+along with the `dotenv <https://github.com/theskumar/python-dotenv>`_ projects
+simplifying your experience, but make sure your environment is set up with Python 3.
+Mesa requires Python3 and does not work in Python 2 environments.
 
-To install Mesa, simply:
+To install Mesa, with pipenv simply clone the repository and run:
```
Comment (Corvince) [inline docs/tutorials/intro_tutorial.rst]: I think this is way too complicated for a intro tutorial. This sentence assumes new users already have pipenv and git installed and know how to/are willing to clone the repository. I think it is fine to recommend pipenv (although I still don't get why we need a pipfile and not just recommend `pipenv install mesa`, see #678 ), but would leave the instructions to install with pip.

File: docs/tutorials/intro_tutorial.rst [hunk 2]
```diff
@@ -24,7 +24,7 @@ model as it runs. Finally, we go over how to write your own
 visualization module, for users who are comfortable with JavaScript.
 
 You can also find all the code this tutorial describes in the
-**examples/Tutorial-Boltzmann\_Wealth\_Model** directory of the Mesa
+`examples/boltzmann_wealth_model/ <https://github.com/projectmesa/mesa/tree/master/examples/boltzmann_wealth_model>`__ directory of the Mesa
 repository.
 
 Sample Model Description
```
Comment (Corvince) [inline docs/tutorials/intro_tutorial.rst]: (Not Part of the PR, but still:)
The literature references (line 35 and line 162 are still broken)

File: docs/tutorials/adv_tutorial.rst [hunk 3]
```diff
@@ -125,19 +118,30 @@ The full code should now look like:
...
 If this doesn't show you the visualization, something may have gone
 wrong with the server launch.)
 
-You should see something like the figure below: the model title, a grid with
-red circles respresenting the agents, and the model controls at the top.
+You should see something like the figure below: the model title, an
+empty space where the grid will be, and a control panel off to the
+right.
+
+.. figure:: files/viz_empty.png
```
Comment (Corvince) [inline docs/tutorials/adv_tutorial.rst]: This is a figure of the old visualization style (before the bootstrap update)

**Comment by jackiekazil** [comment]
@Corvince for the sake of moving this along, I am going to merge, but create a few tickets for corrections. Ty for the review.

**Comment by Corvince** [comment]
Yes, I would have suggested to do so! 

However, the `master` branch on readthedocs doesn't seem to be automatically updated. Can you see on how to enable automatic updates? [This page](https://docs.readthedocs.io/en/stable/webhooks.html) talks about how to do it and at the bottom talks about a change in the system that could have stopped previous automatic updates (if they ever happened for mesa).

**Comment by jackiekazil** [comment]
@Corvince ty for the flag & for the new ticket. Agreed!

File: docs/tutorials/intro_tutorial.rst (hunk 2 above)
Comment (Gigi1111) [reply docs/tutorials/intro_tutorial.rst:30]: After the intro tutorial rewrite in the `doc_builds` branch, this link no longer exists.
